src/Stream/DynamicMemoryWriter.cpp

```diff
@@ -56,4 +56,10 @@ namespace Stream
 	MemoryReader DynamicMemoryWriter::GetReader() {
 		return MemoryReader(streamBuffer.data(), streamBuffer.size());
 	}
+
+	// Clear all currently set memory
+	void DynamicMemoryWriter::Clear() {
+		SeekBeginning();
+		streamBuffer.clear();
+	}
```

Comment:
Would it make sense when DynamicMemoryWriter::GetReader is called to move the data from the writer to a new object that does not share state with the original DynamicMemoryWriter after the move? This seems safer than the current operation and more intuitive to how I would use DynamicMemoryWriter. However, I assume this approach would tradeoff performance that would become more noticeable the larger the data written before handing off to the reader. I'm not sure how successful the move commands in C++ are at mitigating copy commands.

Reply:
Hmm, interesting approach. I suppose a `std::move` of the underlying buffer would be fairly efficient. Though that reader would need to own the buffer. With the current design, the reader only has a view of an external buffer, but doesn't own it. You would need a new reader object with buffer ownership semantics.

Reply:
I propose we close this branch without merging and add an issue to cover the std::move use case of DynamicMemoryWriter with a new reader class. I can remove FileWriter from the BitmapFile test code using the current syntax for now to close the original issue out.

Now that I'm thinking about a Reader that owns its stream, that may be a useful feature in itself. You could select between using the shared memory reader and the sole owner reader based on your use case if std::move is cheap enough. I hadn't thought about it before.

Reply:
Sounds reasonable. I think there have been a few cases where we've considered the idea of a reader that owns its own memory buffer. In this case, having such a reader could allow us to offer safer semantics for reading after filling a buffer with the memory writer class. Probably lots of other places too where it would offer safer or more convenient semantics.